In [1]:
import pandas as pd
from scipy import stats
from statsmodels.stats.anova import AnovaRM


In [2]:
data = pd.read_excel('research_data.xlsx')  
data['Change'] = data['Posttest']-data['Pretest']
data['ID'] = data.index.values
data

Condition  Pretest  Posttest  Change  ID
0           1       50        75      25   0
1           1       40        70      30   1
2           1       55        85      30   2
3           1       49        68      19   3
4           1       60        83      23   4
5           1       47        72      25   5
6           1       52        78      26   6
7           1       53        76      23   7
8           1       48        71      23   8
9           1       51        79      28   9
10          2       52        73      21  10
11          2       53        72      19  11
12          2       51        76      25  12
13          2       50        71      21  13
14          2       54        75      21  14
15          2       62        78      16  15
16          2       42        62      20  16
17          2       49        68      19  17
18          2       55        69      14  18
19          2       57        73      16  19

First we will explore our data. A test of normality on the pretest data will be preformed to see if it is appropriate to use t-tests. We will also compute the mean change in scores under conditions 1 and 2  and check to see there is a significant difference between the pretest groups. A difference between groups in the pretest data would lead to a design flaw as we are trying to see the effects of conditions 1 and 2. Our goal is to see if there is a significant difference in the change of test values. 

In [3]:
Cond_1_shapiro_test = stats.shapiro(data['Pretest'].loc[data['Condition']==1])
Cond_2_shapiro_test = stats.shapiro(data['Pretest'].loc[data['Condition']==2])
Cond_1_shapiro_test,Cond_2_shapiro_test,


((0.9697864651679993, 0.888858437538147),
 (0.9697864651679993, 0.888858437538147))

In [4]:
print('Mean change under conditions 1 and 2  was', data['Change'].loc[data['Condition']==1].mean(axis = 0), 'and',data['Change'].loc[data['Condition']==2].mean(axis = 0))

Mean change under conditions 1 and 2  was 25.2 and 19.2


Both results presented with a p-value of .8889 meaning we can assume the data is normal. We also found condition 1 produces a higher mean change at 25.2 compared to 19.2 Normality implies use of 2-sample t-test is approprite as the assumptions are met. Next, we will test to see if the pretest results are significnatly different between conditions 1 and 2 using a 2 sample welch's t-test. 

In [5]:
Welchs = stats.ttest_ind(data['Pretest'].loc[data['Condition']==1],data['Pretest'].loc[data['Condition']==2],equal_var = False)
Welchs

Ttest_indResult(statistic=-0.8476808799610246, pvalue=0.40774650093191034)

There is no significant difference with a p-value of .4077. Now, we see if there is a difference between the change in the  groups scores. Since we are working with related groups, we will use a repeated measure ANOVA test on the data. We can acheieve the RM-ANOVA test by using a one-way ANOVA test on the pretest and the change data columns.

In [13]:
F,P = stats.f_oneway(data['Pretest'],data['Change'])
print('The value of the F statistic and p-value is' ,F,' and ',P)

The value of the F statistic and p-value is 362.39302377249624  and  4.9929589380519775e-21


With a P values of near 0 , we reject the null hypothesis of equality. In this case, it would imply there is a significant difference between the conditions 1 and 2 on the score of the postest. We can assume the residuals follow a normal distribution as the model would throw an error if they did not. To check the method I will also preform a 2 sample t test using the change column.

In [7]:
Welchs = stats.ttest_ind(data['Change'].loc[data['Condition']==1],data['Change'].loc[data['Condition']==2],equal_var = False)
Welchs

Ttest_indResult(statistic=4.032996434533988, pvalue=0.0007891693843779189)

The 2 sample t-test on the change in value was significant at p=.0008. Thus this test also suggests a significant difference between the groups. Therefore, there is enough evidence to suggest that applying condition 1 will lead to a higher change in test scores compared to applying condition 2. 